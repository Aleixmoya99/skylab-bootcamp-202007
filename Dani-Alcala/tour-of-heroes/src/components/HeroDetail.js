import React from 'react';

class HeroDetail extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			heroName: 'Bombasto',
			heroId: 14,
		};
		this.onFieldChange = this.onFieldChange.bind(this);//le asigno el contexto que me interesa, que es el objeto que se crea con esta clase. El form tiene su propio contexto
	}
	onFieldChange(event) {
		this.setState({
			// heroName: event.target.value //es pot fer aixi, més fàcil
			[event.target.name]: event.target.value //propiedad dinamica, així està preparat per si poso un altre input amb el name associat correctament a this.state
		})
		console.log(this.state.heroName)
	}

	render() {
		return (
			<form>
				<p>id: {this.state.heroId}</p>
				<label htmlFor="heroName">
					name:
					<input
						name="heroName"
						placeholder="Hero name"
						value={this.state.heroName}
						onChange={this.onFieldChange}
					/>
				</label>
			</form>
		);
	}
}

export default HeroDetail;
