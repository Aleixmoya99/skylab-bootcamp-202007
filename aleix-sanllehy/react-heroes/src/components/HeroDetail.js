import React from 'react';
import heroData from './hero.mock';

class HeroDetail extends React.Component {
	constructor(props) {
		super(props);
		this.state = this.getId();
		//this.state = { heroId: null, heroName: '' };
		this.onFieldChange = this.onFieldChange.bind(this);
	}

	onFieldChange(myEvent) {
		this.setState({ name: myEvent.target.value });
		//this.setState({ [myEvent.target.name]: myEvent.target.value });
		//console.log(myEvent.target.name, '=>', myEvent.target.value);
	}

	getId() {
		const url = window.location.pathname;
		const urlId = +url.split('/')[2];
		return heroData.find((element) => element.id === urlId);
	}

	render() {
		return (
			<main className="hero-details">
				<h3>
					<span>{this.state.name}</span> details!
				</h3>
				<form>
					<p className="hero-details__element">
						<span className="hero-details__id">id:</span>
						<span className="hero-details__id">{this.state.id}</span>
					</p>
					<label className="hero-details__name" htmlFor="heroName">
						name:
						<input
							name="heroName"
							placeholder="Hero name"
							value={this.state.name}
							onChange={this.onFieldChange}
						/>
					</label>
				</form>
				<span className="id btn btn-secondary">Back</span>
			</main>
		);
	}
}

export default HeroDetail;
