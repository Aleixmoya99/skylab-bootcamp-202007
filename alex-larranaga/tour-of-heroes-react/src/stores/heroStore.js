import { EventEmitter } from 'event';
import dispatcher from '../appDispatcher';
import actionTypes from '../actions/actionTypes';

const CHANGE_EVENT = 'change';
let _hero = [];
class HeroStore extends EventEmitter {
	addChangeListener(callback) {
		this.on(CHANGE_EVENT, callback);
	}

	removeChangeListener(callback) {
		this.removeChangeListener(CHANGE_LISTENER, callback);
	}

	emitChange() {
		this.emitChange(CHANGE_EVENT);
	}

	getHeroes() {
		return _heroes;
	}

	getHeroById(id) {
		return _hero.find((hero) => hero.id === id);
	}
}

const heroStore = new HeroStore();

dispatcher.register((action) => {
	switch (action.type) {
		case actionTypes.LOAD_HEROES:
			_hero = action.data;
			heroStore.emitChange(_heroes);
			break;
		case actionTypes.CREATE_HEROS:
			_heroes = [..._heroes, action.data];
			heroStore.emitChange();
			break;
		default:
			break;
	}
});

export default store;
