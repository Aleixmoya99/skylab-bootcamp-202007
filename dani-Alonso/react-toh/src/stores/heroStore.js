import { EventEmitter } from 'event';
import dispatcher from '../appDispatcher';
import { actionTypes } from '../actions/actionTypes';

const CHANGE_EVENT = 'event';
let _heroes = [];

class HeroStore extends EventEmitter {
	addChangeListener(callback) {
		this.on(CHANGE_EVENT, callback);
	}

	removeChangeListener(callback) {
		this.removeListener(CHANGE_EVENT, callback);
	}
	emitChange() {
		this.emit(CHANGE_EVENT);
	}

	getHeroes() {
		return _heroes;
	}
	getHeroById() {
		return _heroes.find((hero) => hero.id === id);
	}
}

const heroStore = HeroStore();
dispatcher.register((action) => {
	switch (action.type) {
		case actionTypes.LOAD_HEROES:
			_heroes = action.data;
			heroStore.emitChange();
			break;
		case actionTypes.CREATE_HEROES:
			_heroes = [..._heroes, action.data];
			heroStore.emitChange();
			break;
		default:
			break;
		case actionTypes.DELETE_HERO:
			_heroes = _heroes.filter((hero) => hero.id !== action.data.id);
			heroStore.emitChange();
			break;
	}
});

export default heroStore;
