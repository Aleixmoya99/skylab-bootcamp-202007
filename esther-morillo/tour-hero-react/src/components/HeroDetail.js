import React from 'react'


function HeroDetail({ id, name }) {
    return (
        <>
            <div>
                <h2>{name} details!</h2>
                <p>id:<span>{id}</span></p>
                <label>name:<input></input></label>
            </div>
        </>
    );
}

export default HeroDetail;