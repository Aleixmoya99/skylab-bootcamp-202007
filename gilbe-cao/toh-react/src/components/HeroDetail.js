import React from 'react';
import heroList from '../hero.mock';
import { Prompt } from 'react-router-dom';

class HeroDetail extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			heroName: '',
			heroId: null,
			isFormDirty: false
		};

		this.onFieldChange = this.onFieldChange.bind(this);
	}

	onFieldChange(myEvent) {
		this.setState({
			[myEvent.target.name]: myEvent.target.value,
			isFormDirty: true
		});
	}

	getHeroById(id) {
		return heroList.find((hero) => hero.id === id);
	}

	componentDidMount() {
		const hero = this.getHeroById(+this.props.match.params.heroId);
		this.setState({
			heroName: hero.name,
			heroId: hero.id
		});
	}

	render() {
		return (
			<form>
				<p>id: {this.state.heroId}</p>
				<label htmlFor="heroName">
					name:
					<input
						name="heroName"
						placeholder="Hero name"
						value={this.state.heroName}
						onChange={this.onFieldChange}
					/>
				</label>
				<Prompt
					when={this.state.isFormDirty}
					message="Are you sure you want to navigate away?"
				/>
			</form>
		);
	}
}

export default HeroDetail;
