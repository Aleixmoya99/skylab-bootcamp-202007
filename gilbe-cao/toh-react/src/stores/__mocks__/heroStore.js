let __heroes = [];
export default {
	_setHeroes(heroes) {
		__heroes = heroes;
	},

	getHeroes() {
		return __heroes;
	},

	getHeroById(id) {
		return __heroes.find((hero) => hero.id === id);
	},

	addChangeListener(callback) {},

	removeChangeListener(callback) {},

	emitChange() {}
};
