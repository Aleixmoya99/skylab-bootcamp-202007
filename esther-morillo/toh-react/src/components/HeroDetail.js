import React from 'react';
import heroList from '../hero.mock';
import heroStore from '../stores/heroStore';
import { Prompt, Link } from 'react-router-dom';

class HeroDetail extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			heroName: '',
			heroId: null,
			formIsDirty: true
		};

		this.onFieldChange = this.onFieldChange.bind(this);
	}

	onFieldChange(myEvent) {
		this.setState({
			[myEvent.target.name]: myEvent.target.value,
		});
	}

	

	getHeroById(id) {
		return heroList.find((hero) => hero.id === id);
	}

	componentDidMount() {
		const hero = this.getHeroById(+this.props.match.params.heroId);
		this.setState({
			heroName: hero.name,
			heroId: hero.id
		});
	}

	render() {
		return (
			<>
			<form>
				<p>id: {this.state.heroId}</p>
				<label htmlFor="heroName">
					name:
					<input
						name="heroName"
						placeholder="Hero name"
						value={this.state.heroName}
						onChange={this.onFieldChange}
					/>
				</label>
				<Prompt
					when={this.state.formIsDirty}
					message="Are you sure you want to navigate away?"
				/>
			</form>

			<Link 
			to={`/hero/`}
			onClick={this.getHeroNewName}
			>
					Back
			</Link>
			</>
		);
	}
}

export default HeroDetail;
