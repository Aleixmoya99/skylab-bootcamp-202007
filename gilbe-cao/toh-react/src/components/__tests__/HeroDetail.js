import React from 'react';
import renderer from 'react-test-renderer';
import HeroDetail from '../HeroDetail';
import { BrowserRouter } from 'react-router-dom';
import heroList from '../../hero.mock';
import heroStore from '../../stores/heroStore';

describe('HeroDetail', () => {
	let heroDetailTree;
	let instance;
	let component;
	let text;
	let props = {
		match: {
			params: {
				heroId: null
			}
		}
	};

	beforeEach(async () => {
		heroDetailTree = renderer.create(
			<BrowserRouter>
				<HeroDetail {...props} />
			</BrowserRouter>
		);
		instance = heroDetailTree.root;
		component = instance.findByType('h2');
		text = component.children[0];
		heroDetailTree.update();
	});

	it('should match with hero ID', async () => {
		expect(heroDetailTree).toMatchSnapshot();
	});

	it('should display a title without id', async () => {
		expect(text).toEqual('Register a new hero:');
	});
});
