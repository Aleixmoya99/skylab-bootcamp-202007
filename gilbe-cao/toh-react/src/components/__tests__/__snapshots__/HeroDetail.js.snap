// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroDetail should match with hero ID 1`] = `null`;
